src/sinks/prometheus/exporter.rs

```diff
@@ -343,6 +375,11 @@ fn handle(
 ) -> Response<Body> {
     let mut response = Response::new(Body::empty());
 
+    if !check_auth(&req, auth) {
+        *response.status_mut() = StatusCode::UNAUTHORIZED;
+        return response;
```

Comment:
Do we want to add a `WWW-Authenticate` header here too? Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401

Reply:
Yeah, I think we should add one HTTP header `WWW-Authenticate` with a value `Basic, Bearer`. Multiple challenges in one headers are allowed by the specification.

Reply:
Fixed